WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -302,15 +403,31 @@ private extension ProductImageUploader {
                                                             productOrVariationID: key.productOrVariationID,
                                                             error: .failedUploadingImage(asset: asset, error: error))
                 if statusUpdatesExcludedProductKeys.contains(key) == false {
-                    errorsSubject.send(infoError)
+                    if featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) {
+                        let failedStatus = ProductImageStatus.uploadFailure(asset: asset,
+                                                                            error: error,
+                                                                            siteID: key.siteID,
+                                                                            productID: key.productOrVariationID)
+                        self.userDefaultsStatuses.updateStatus(failedStatus)
+                    } else {
+                        errorsSubject.send(infoError)
+                    }
                 }
             }
         }
         imageUploadSubscriptions.insert(observationToken)
     }
 
     func removeProductFromActiveUploads(key: Key) {
-        activeUploadsPublisher.removeAll(where: { $0 == key })
+        if featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) {
+            userDefaultsStatuses.removeStatus(where: { status in
+                status.siteID == key.siteID &&
+                status.productOrVariationID == key.productOrVariationID &&
+                status.isUploading
+            })
+        } else {
+            activeUploadsPublisher.removeAll(where: { $0 == key })
+        }
```

Comment:
Save comment as above, since you're maintaining the storage disregarding of the feature flag, you should remove the feature flag check here.

Reply:
As mentioned above, I kept this logic behind the feature flag because otherwise, it means writing states in User Defaults regardless of the feature flag, which is unnecessary (and it's just an if-else condition). The change previously suggested was mostly done to avoid unwrapping an optional value.
Wdyt?

Reply:
I suppose we'll remove the code on the other side of the feature flag checks, so keeping this for now should be fine 👍